logic/scripts/drops.py

```diff
@@ -0,0 +1,82 @@
+from contextlib import closing
+import json
+import webbrowser
+import re
+
+from bs4 import BeautifulSoup
+
+import requests
+from requests.exceptions import RequestException
+from tools import db_utils
+from util import tasks
+from config import constants
+from types import SimpleNamespace
+from datetime import date, datetime, timezone
+from dateutil import parser
+
+
+
+headers = {
+    "User-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36"
+}
+
+def sort_drops(drops):
+    sorted_drops = sorted(drops, key=lambda x: x.startDate, reverse=True)
+    return sorted_drops
+
+def parse_date(date):
+    return parser.parse(date)
+
+def filter_upcoming_drops(drops):
+    upcomingDrops = []
+    for drop in drops:
+        if not drop.startDate or not drop.endDate:
+            continue
+        if (parse_date(drop.startDate) > datetime.now(timezone.utc) or parse_date(drop.endDate) > datetime.now(timezone.utc)):
+            drop.countdown = parse_date(drop.startDate).strftime('%A %B %m')
+            upcomingDrops.append(drop)
+    upcomingDrops = sort_drops(upcomingDrops) 
+    
+    return upcomingDrops      
+
+def filter_past_drops(drops, allPast):
+    pastDrops = []
+    for drop in drops:
+        if not drop.endDate:
+            continue
+        if parse_date(drop.endDate) < datetime.now(timezone.utc):
+            pastDrops.append(drop) 
+    pastDrops = sort_drops(pastDrops)
+
+    print(allPast)
+    print(len(pastDrops))
+
+    if allPast == 'true':
+        return pastDrops
+    else:    
+        return pastDrops[slice(0, 3)]  
+               
+
+def get_drops(allPast):
+
+    headers = {
+    "Content-Type": "application/json",
+    }
+    url = "{0}/site-marketing".format(constants.LAUNCHPAD_API)
+
+    drops = []
+    try:
+        with closing(requests.get(url, headers=headers)) as resp:
```

Comment:
The website can get under pretty heavy load. Especially the home page. Which will translate in high traffic on the NFT API server. I'd definitely recommend considering caching the data. And the data changes very infrequently (only when a new drop gets added), so it is a good candidate for applying caching. We have examples in this codebase for caching either in memory or in Redis. It is pretty straight forward.

Reply:
I implemented an `in memory` cache. requested review to make sure it made sense to you. seems to be working for me. redis serialization was causing me problems...